src/authorizations/components/redesigned/ResourceAccordion.tsx

```diff
@@ -0,0 +1,154 @@
+        buckets: props.bucketPermissions,
+      },
+    }
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    if (!props.telegrafPermissions) {
+      return null
+    }
+    return {
+      ...state,
+      permissions: {
+        telegrafs: props.telegrafPermissions,
+        buckets: props.bucketPermissions,
+      },
+    }
+  }
+
+  public render() {
+    const {resources} = this.props
+    const {permissions} = this.state
+
+    if (resources) {
+      return resources.map(resource => {
+        const resourceName =
+          resource.charAt(0).toUpperCase() + resource.slice(1)
+        return (
+          <Accordion key={resource}>
+            <ResourceAccordionHeader
+              resourceName={resourceName}
+              permissions={permissions[resource]}
+              onToggleAll={this.handleToggleAll}
+            />
+            <GetResources resources={[ResourceType[resourceName]]}>
+              {resourceName === 'Telegrafs' ? (
+                <ResourceAccordionBody
+                  resourceName={resource}
+                  permissions={permissions[resource]}
+                  onToggle={this.handleIndividualToggle}
+                  title="Individual Telegraf Configuration Names"
+                />
+              ) : (
+                <ResourceAccordionBody
+                  resourceName={resource}
+                  permissions={permissions[resource]}
+                  onToggle={this.handleIndividualToggle}
+                  title="Individual Bucket Names"
+                />
+              )}
+            </GetResources>
+          </Accordion>
+        )
+      })
+    }
+    return null
+  }
+
+  handleToggleAll = (resourceName, permission, value) => {
+    const {permissions} = this.state
+    const newPerm = {...permissions}
+    const name = resourceName.toLowerCase()
+    Object.keys(newPerm[name]).map(key => {
+      newPerm[name][key].permissions[permission] = !value
+    })
+    this.setState({
+      permissions: newPerm,
+    })
+  }
+
+  handleIndividualToggle = (resourceName, id, permission) => {
+    const {permissions} = this.state
+
+    const permValue = permissions[resourceName][id].permissions[permission]
+
+    const newPerm = {...permissions}
+    newPerm[resourceName][id].permissions[permission] = !permValue
+
+    this.setState({
+      permissions: newPerm,
+    })
+  }
+}
+
+const mstp = (state: AppState) => {
+  const telegrafs = getAll<Telegraf>(state, ResourceType.Telegrafs)
+  const telegrafPermissions = {}
+  telegrafs.forEach(t => {
+    telegrafPermissions[t.id] = {
+      id: t.id,
+      orgID: t.orgID,
+      name: t.name,
+      permissions: {read: false, write: false},
+    }
+  })
+
+  const buckets = getAll<Bucket>(state, ResourceType.Buckets)
+  const bucketPermissions = {}
+  buckets.forEach(b => {
+    bucketPermissions[b.id] = {
+      id: b.id,
+      orgID: b.orgID,
+      name: b.name,
+      permissions: {read: false, write: false},
+    }
+  })
```

Comment:
what do you think about changing `t` to ~`token`~ `telegraf` and `b` to `bucket`?